src/overlays/actors/ovl_En_Fish/z_en_fish.c

```diff
@@ -47,73 +63,709 @@ static ColliderJntSphInit D_80A17040 = {
+        EnFish_Respawning_SetupApproachPlayer(this);
+    }
+}
+
+void EnFish_Respawning_SetupApproachPlayer(EnFish* this) {
+    this->actor.gravity = 0.0f;
+    this->actor.minVelocityY = 0.0f;
+    EnFish_SetInWaterAnimation(this);
+    this->timer = Rand_S16Offset(10, 40);
+    this->unk_250 = 0;
+    this->actionFunc = EnFish_Respawning_ApproachPlayer;
+}
+
+void EnFish_Respawning_ApproachPlayer(EnFish* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Player* player = PLAYER;
+    s32 pad2;
+    Vec3f sp38;
+    s16 yaw;
+    s16 temp_a0_2;
+
+    EnFish_SetYOffset(this);
+    Math_SmoothStepToF(&this->actor.speedXZ, 1.8f, 0.1f, 0.5f, 0.0f);
+
+    if (EnFish_XZDistanceSquared(&this->actor.world.pos, &this->actor.home.pos) > SQ(80.0f)) {
+        yaw = Math_Vec3f_Yaw(&this->actor.world.pos, &this->actor.home.pos);
+        Math_StepToAngleS(&this->actor.world.rot.y, yaw, 3000);
+    } else {
+        if ((s16)globalCtx->state.frames & 0x40) {
+            temp_a0_2 = (this->actor.yawTowardsPlayer + 0x9000);
+        } else {
+            temp_a0_2 = (this->actor.yawTowardsPlayer + 0x7000);
+        }
+
+        sp38.x = player->actor.world.pos.x + (Math_SinS(temp_a0_2) * 20.0f);
+        sp38.y = player->actor.world.pos.y;
+        sp38.z = player->actor.world.pos.z + (Math_CosS(temp_a0_2) * 20.0f);
+
+        yaw = Math_Vec3f_Yaw(&this->actor.world.pos, &sp38);
+        Math_StepToAngleS(&this->actor.world.rot.y, yaw, 3000);
+    }
+
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    this->skelAnime.playSpeed = CLAMP_MAX((this->actor.speedXZ * 1.5f) + 0.8f, 4.0f);
+
+    SkelAnime_Update(&this->skelAnime);
+
+    if (this->timer <= 0) {
+        EnFish_Respawning_SetupSlowDown(this);
+    }
+}
+
+// Dropped type functions
+
+void EnFish_Dropped_SetupFall(EnFish* this) {
+    this->actor.gravity = -1.0f;
+    this->actor.minVelocityY = -10.0f;
+    this->actor.shape.yOffset = 0.0f;
+    EnFish_SetOutOfWaterAnimation(this);
+    this->unk_250 = 5;
+    this->actionFunc = EnFish_Dropped_Fall;
+    this->timer = 300;
+}
+
+void EnFish_Dropped_Fall(EnFish* this, GlobalContext* globalCtx) {
+    Math_SmoothStepToF(&this->actor.speedXZ, 0.0f, 0.1f, 0.1f, 0.0f);
+    Math_StepToAngleS(&this->actor.world.rot.x, 0x4000, 100);
+    Math_StepToAngleS(&this->actor.world.rot.z, -0x4000, 100);
+    this->actor.shape.rot.x = this->actor.world.rot.x;
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    this->actor.shape.rot.z = this->actor.world.rot.z;
+    SkelAnime_Update(&this->skelAnime);
+
+    if (this->actor.bgCheckFlags & 1) { // On floor
+        this->timer = 400;
+        EnFish_Dropped_SetupFlopOnGround(this);
+    } else if (this->actor.bgCheckFlags & 0x20) { // In water
+        EnFish_Dropped_SetupSwimAway(this);
+    } else if ((this->timer <= 0) && (this->actor.params == FISH_DROPPED) && (this->actor.floorHeight < -31990.0f)) {
+        osSyncPrintf(VT_COL(YELLOW, BLACK));
```

Comment:
i guess -31990 is `BGCHECK_Y_MIN + 10.0f`

Reply:
Whoever wrote Fish and Insect seemed to like this number as their min, rather than the actual BGCHECK number.

Reply:
yea, id say to still write it in terms of the actual minimum